src/wallet/wallet.cpp

```diff
@@ -5040,12 +5044,12 @@ bool CWalletTx::IsFromMe(const isminefilter& filter) const
     return false;
 }
 
-bool CWalletTx::IsTrusted() const
+bool CWalletTx::IsTrusted(const std::optional<int>& asOfHeight) const
```

Comment:
Line 5060 onward: does it make sense to consider mempool inputs from us to be trusted when `asOfHeight` is set (and is less than the current known height)? Current mempool transactions may depend on outputs of transactions mined after `asOfHeight`.

Reply:
Local behavior now is to treat a `minconf` of 0 as 1 if `asOfHeight` is also provided. However, this could instead be an RPC error. Since `asOfHeight` is a new parameter, this wouldn’t break any previously-working queries.

Reply:
The argument for not making it error is that we’re just ignoring the mempool, so 0 is still a valid `minconf`, but the mempool is “empty”.

Reply:
@sellout and I discussed this and decided that it should be an error if minconf == 0 when `asOfHeight` is set.